pkg/oauth/apis/oauth/validation/validation_test.go

```diff
@@ -746,3 +746,38 @@ func TestValidateAuthorizeTokensUpdate(t *testing.T) {
 		}
 	}
 }
+
+func TestValidateUserNameField(t *testing.T) {
+	fieldArg := field.NewPath("metadata", "name")
+	valid := "kube:admin"
+	errs := ValidateUserNameField(valid, fieldArg)
+	if len(errs) != 0 {
+		t.Errorf("expected success: %v", errs)
+	}
+
+	errorCases := map[string]struct {
+		name string
+		T    field.ErrorType
+		F    string
+	}{
+		"system:admin": {
+			T: field.ErrorTypeInvalid,
+			F: "metadata.name",
+		},
+	}
+	for k, v := range errorCases {
+		errs := ValidateUserNameField(k, fieldArg)
+		if len(errs) == 0 {
+			t.Errorf("expected failure %s for %v", k, k)
+			continue
+		}
+		for i := range errs {
+			if errs[i].Type != v.T {
+				t.Errorf("%s: expected errors to have type %s: %v", k, v.T, errs[i])
+			}
+			if errs[i].Field != v.F {
+				t.Errorf("%s: expected errors to have field %s: %v", k, v.F, errs[i])
```

Comment:
suggestion: `t.Errorf("expected error field %s, got %s", tc.expected_err_field, errs[i].Field)`

Reply:
The same as the above.

Reply:
I refactored the test using the latest template from Goland. If it's not sufficient, please point me out what shall I use instead.